### DISCUSSION TITLE
Ask questions, get answers!

### DISCUSSION THREAD

**Discussion by ebajcar** [ebajcar/web10199_worksheets#5] [Q&A]
📢 **Ask questions** and get answers from classmates and instructions.

**Comment by ebajcar** [comment]
### If you need to review or catch-up from previous semester

- [SYST10049 material](https://ebajcar.github.io/modules_syst10049/)
- [Planning a simple web app](https://ebajcar.github.io/modules_syst10049/pages/ch_2_3_1.html)
-  [Planning and designing a web app](https://ebajcar.github.io/modules_syst10049/pages/ch_2_4_1.html)
- [Writing documentation and coding](https://ebajcar.github.io/modules_syst10049/pages/ch_2_4_2.html)

**Reply by nofalab** [reply]
Hi Professor,

I have question about set3/primitives_strings.html exercise, what is the difference between a or b and c below?

Thank you

Abdallah Nofal

var name = "Ellen";
var a = (name)
  ? "Hello, ${name}!"
  : "the result is ${1 + 2}";
console.log("a: ", a);

var b = (name)
  ? 'Hello, ${name}!'
  : 'the result is ${1 + 2}';
console.log("b: ", b);

var c = (name)
  ? `Hello, ${name}!`
  : `the result is ${1 + 2}`;
console.log("c: ", c);

var d = (name === "John")
  ? `Hello, ${name}!`
  : `the result is ${1 + 2}`;
console.log("d: ", d);

**Reply by ebajcar** [reply]
> Hi Professor,
> 
> I have question about set3/primitives_strings.html exercise, what is the difference between a or b and c below?
> 
> Thank you
> 
> Abdallah Nofal
> 
> var name = "Ellen";
> var a = (name)
> ? "Hello, ${name}!"
> : "the result is ${1 + 2}";
> console.log("a: ", a);
> 
> var b = (name)
> ? 'Hello, ${name}!'
> : 'the result is ${1 + 2}';
> console.log("b: ", b);
> 
> var c = (name)
> ? `Hello, ${name}!`
> : `the result is ${1 + 2}`;
> console.log("c: ", c);
> 
> var d = (name === "John")
> ? `Hello, ${name}!`
> : `the result is ${1 + 2}`;
> console.log("d: ", d);

If I understand your question correctly, we are just creating different variables to identify which result belongs to which statement, so that you can compare the result. There is no other significance to choosing those variable names.

**Reply by nofalab** [reply]
The output in console from a & b & c are different in the exercise, and regardless of variable names, the results should match as I don't see difference in syntax. Unless I am missing something here, 
thank you

**Reply by ebajcar** [reply]
**a is using double quotes, b is using single quotes, and c is using back quotes.**

**Comment by xiaoyazz** [comment]
Hi professor,

I just opened assignment 3 and there is no content inside. I reloaded the page several times but it didn't work. Then I closed the window and tried to email you on slate, but there is an alert saying I'm on a quiz and the email function has been disabled. Do I just click the submit button?
Thank you.
Xiaoya
![Capture](https://user-images.githubusercontent.com/84748829/120691662-ba8d3980-c474-11eb-8135-dd7ba6f53734.JPG)

**Reply by ebajcar** [reply]
New information available in SLATE.

**Reply by xiaoyazz** [reply]
Thank you!

**Comment by nofalab** [comment]
Hi Professor; 

Is it okay to try writing the games JS codes using one's own logic and algorithm, provided that it gives the same results and functionality. If yes then what I need to add is more of commenting than citing, right?

Also, is there a definitive way to measure how efficient the code I write is compared to other scripts ( i.e. the code that has less processing ticks is the more efficient)?

Thank you 

Abdallah Nofal

**Reply by ebajcar** [reply]
> Hi Professor;
> 
> Is it okay to try writing the games JS codes using one's own logic and algorithm, provided that it gives the same results and functionality. If yes then what I need to add is more of commenting than citing, right?
> 
> Also, is there a definitive way to measure how efficient the code I write is compared to other scripts ( i.e. the code that has less processing ticks is the more efficient)?
> 
> Thank you
> 
> Abdallah Nofal

There is no definitive way to measure - it is a complex topic; similar to trying to answer "is a definitive programming language". The best approach is to learn about the principles, conventions, best practices associated with developing solutions; explore deeply (read/listen a lot), consider alternative solutions, analyze and evaluation your observations.

**Reply by nofalab** [reply]
Clear, thank you

**Comment by E-Annie** [comment]
Hello Professor Ellen, 

for assignment 2, could we make the javascript and css external instead of internal? I have linked all the external links to the HTML and published them as well.

Thank you, 
Annie Kim

**Comment by ebajcar** [comment]
For this assignment, Assignment 2, follow the dropbox instructions, keep your additional CSS and Javascript internal.
Ellen
________________________________

**Comment by zeidomar** [comment]
Hello Professor Ellen, I was wondering when will assignment 4 will be due and where can I find the instructions for the assignment.

**Reply by ebajcar** [reply]
The dropbox is open and available in SLATE.

**Comment by xiaoyazz** [comment]
Dear Professor Bajcar,

For the tic-tac-toe game, are we required to use a similar design/algorithm with your example, or we should use a totally different one? 

Thank you.

Xiaoya

**Reply by ebajcar** [reply]
Whatever demonstrates what you have learned and complies with the concepts and requirements of this course.

**Reply by xiaoyazz** [reply]
Thank you!

**Comment by xiaoyazz** [comment]
Dear Professor Bajcar,

I have a question about the URL. For assignment 2, I didn't put the page of the RPSLS game in the "content" folder. This time I have reorganized my file folders on cPanel and will put all the games pages in the "content" folder as you required. Please advise if I need to submit a new URL of assignment 2 to you. Thank you.

Best regards,

Xiaoya

**Comment by zeidomar** [comment]
Hello Professor Bajcar,

I was just wondering where can get the game rules and description for casino craps? cuz I am having trouble understanding the game online.

omar zeid.

**Reply by nofalab** [reply]
Hi, 
Everything u need is in a short video inside week4 slides..
Hope this helps
